### ISSUE TITLE
Context Menus Misbehaving

### ISSUE THREAD

**Issue by kevincox** [mumble-voip/mumble#6145]
### Description

When clicking context menus for channels about 19/20 times the result is for the root channel. When clicking the context menu for users 19/20 times nothing happens (maybe because it is trying to run a user action on the root channel?). (However mute-self and deafen-self seem to work reliably).

Possibly related is that dragging doesn't work. Whenever I start dragging a white pixel to two appear at the top left corner of the root channel.

Not only is this a UX disaster but it could also be a security concern as it is very easy to accidentally edit the ACLs of the root channel instead of the intended channel if you don't notice the title of the window has the wrong channel.

### Steps to reproduce

Failing to act on channel:
[Screencast from 2023-06-07 09-00-28.webm](https://github.com/mumble-voip/mumble/assets/494012/e682bc21-bd30-47dc-a01e-debb9ccc0591)

Failing to act on user:
[Screencast from 2023-06-07 08-55-31.webm](https://github.com/mumble-voip/mumble/assets/494012/45229cb8-46fd-44ad-b683-2c80fb232d61)

### Mumble version

1.4.0

### Mumble component

Client

### OS

Linux

### Reproducible?

Yes

### Relevant log output

```shell
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
<D>2023-06-07 09:00:28.830 PulseAudio 16.1.0 from libpulse.so
<X>2023-06-07 09:00:28.833 PipeWire 0.3.71 from libpipewire-0.3.so
<D>2023-06-07 09:00:28.833 JACK 3.0.0.0 (using PipeWire 0.3.71) from libjack.so
<D>2023-06-07 09:00:28.844 libopus 1.3.1 from libopus.so.0
<W>2023-06-07 09:00:28.844 CELT bitstream 8000000b from /nix/store/s60yhc0r2w6mkf3l2cvvrf3nvcnn3h3c-mumble-1.4.287/lib/mumble/libcelt0.so.0.7.0
<W>2023-06-07 09:00:28.844 Theme: "Mumble"
<W>2023-06-07 09:00:28.844 Style: "Dark"
<W>2023-06-07 09:00:28.844 --> qss: ":themes/Mumble/Dark.qss"
<W>2023-06-07 09:00:28.845 Locale is "en_CA" (System: "en_CA")
<W>2023-06-07 09:00:28.862 Database SQLite: "3.41.2"
<W>2023-06-07 09:00:28.878 GlobalShortcutX: Using XI2 2.4
<W>2023-06-07 09:00:28.901 QMetaObject::connectSlotsByName: No matching signal for on_qtvUsers_customContextMenuRequested(QPoint,bool)
<W>2023-06-07 09:00:28.945 SocketRPC: Removing old socket on "/run/user/1000/MumbleSocket"
<W>2023-06-07 09:00:29.112 AudioInput: Opus encoder set for low delay
<W>2023-06-07 09:00:29.112 AudioInput: 72000 bits/s, 48000 hz, 480 sample
<W>2023-06-07 09:00:29.114 AudioInput: Initialized mixer for 1 channel 48000 hz mic and 0 channel 48000 hz echo
<W>2023-06-07 09:00:29.114 AudioOutput: Initialized 2 channel 48000 hz mixer
<W>2023-06-07 09:00:29.135 Updating application palette
<W>2023-06-07 09:00:30.515 Database SQLite: "3.41.2"
<W>2023-06-07 09:00:30.515 OpenSSL Support: 1 (OpenSSL 3.0.8 7 Feb 2023)
<W>2023-06-07 09:00:30.568 ServerHandler: TLS cipher preference is "TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:AES256-SHA:AES128-SHA"
<W>2023-06-07 09:00:32.762 AudioInput: Noise canceller disabled
warning: The VAD has been replaced by a hack pending a complete rewrite
<W>2023-06-07 09:00:33.399 QMetaObject::connectSlotsByName: Connecting slot on_qcbACLUser_activated() with the first of the following compatible signals: ("activated(int)", "activated(QString)")
<W>2023-06-07 09:00:36.645 QMetaObject::connectSlotsByName: Connecting slot on_qcbACLUser_activated() with the first of the following compatible signals: ("activated(int)", "activated(QString)")
<W>2023-06-07 09:00:38.467 QMetaObject::connectSlotsByName: Connecting slot on_qcbACLUser_activated() with the first of the following compatible signals: ("activated(int)", "activated(QString)")
<W>2023-06-07 09:00:40.121 QMetaObject::connectSlotsByName: Connecting slot on_qcbACLUser_activated() with the first of the following compatible signals: ("activated(int)", "activated(QString)")
<W>2023-06-07 09:00:41.673 QMetaObject::connectSlotsByName: Connecting slot on_qcbACLUser_activated() with the first of the following compatible signals: ("activated(int)", "activated(QString)")
<W>2023-06-07 09:00:42.922 QMetaObject::connectSlotsByName: Connecting slot on_qcbACLUser_activated() with the first of the following compatible signals: ("activated(int)", "activated(QString)")
<W>2023-06-07 09:00:44.114 QMetaObject::connectSlotsByName: Connecting slot on_qcbACLUser_activated() with the first of the following compatible signals: ("activated(int)", "activated(QString)")
```

**Comment by kevincox** [comment]
I found that you can work around this by setting the environment variable `QT_QPA_PLATFORM=wayland`. So this is possibly X11/XWayland related.

**Comment by Hartmnt** [comment]
Hi,

Sadly, I can not reproduce this on X11 or in my test VM which is running Ubuntu 22.04 LTS with wayland. Even when running Mumble through Xwayland in the VM. However, I was trying the 1.5.x series.

I think the relevant code passage is:
https://github.com/mumble-voip/mumble/blob/master/src/mumble/MainWindow.cpp#L801

Given your description with the white pixels, I assume that ``qpContextPosition`` contains something weird and ``idx`` resolves to the wrong target.
As we are using Qt methods here and this is not easily reproducible for me, I assume this is an upstream problem in either your specific version of Xwayland or Qt.

Could you please tell us the Qt version and the Xwayland version you are using?
You could also try the 1.5.x series and see, if that magically fixes the problem.

**Comment by kevincox** [comment]
I'm running on NixOS.

- Qt 5.15.9
- XWayland 23.1.1

It seems strange though as I am not seeing any issue on other apps. But maybe there is a slight difference with how they manage context menus that doesn't trigger a Qt bug?

I also have multiple displays with different scaling. However this problem seems to appear on all of them (but maybe the presence of multiple displays is part of the problem).

**Comment by kevincox** [comment]
I tried disabling all but a 100% scale display and still saw this issue, so probably not HiDPI or multi-scale related.

**Comment by kevincox** [comment]
Good guess. I can also reproduce this on VLC. The context menu on the playlist is very flaky with default settings. It took some struggling to get it to use Wayland but after unsetting a bunch of GNOME-related env vars so that it didn't realize it was running on GNOME anymore it used Wayland the the context menu worked properly.

So I guess this is an upstream issue with XWayland or Qt.

**Comment by Hartmnt** [comment]
Wow, cool. Thanks for verifying that. No need to install another test system then. In that case I will close the issue. Feel free to open it again, if the problem is somehow related to how we use the APIs after all.

**Comment by ppfeufer** [comment]
Yep, that's a Qt bug in Wayland sessions.

> Wrong positioning of context menu in multiple screens; fixed in Qt 6.4

- https://community.kde.org/Plasma/Wayland_Showstoppers
- https://bugs.kde.org/show_bug.cgi?id=436016
- https://bugreports.qt.io/browse/QTBUG-85297

**Comment by kevincox** [comment]
I searched for bugs but couldn't find anything. In this case the menus were positioned correctly but for some reason the app seems to think that I clicked a different item. Also none of those mention XWayland, and this works fine for me on native Wayland.
